src/dirtyfields/compare.py

```diff
@@ -0,0 +1,36 @@
+import datetime
+import pytz
+import warnings
+
+from django.conf import settings
+from django.utils import timezone
+
+
+def raw_compare(cls, new_value, old_value):
+    return new_value != old_value
+
+
+def time_zone_support_compare(cls, new_value, old_value):
+
+    if isinstance(new_value, datetime.datetime) and isinstance(old_value, datetime.datetime):
+        db_value_is_aware = timezone.is_aware(old_value)
+        in_memory_value_is_aware = timezone.is_aware(new_value)
+
+        if db_value_is_aware != in_memory_value_is_aware:
```

Comment:
Cosmetics: I personally prefer returning early from the function instead of creating deep nesting. In this case: `db_value_is_aware != in_memory_value_is_aware: return new_value != old_value`